test/e2e/framework/framework.go

```diff
@@ -206,20 +117,106 @@ func KubernetesClientEnv(t *testing.T) []string {
 	return []string{"KUBERNETES_SERVICE_HOST=" + host, "KUBERNETES_SERVICE_PORT=" + port}
 }
 
+type Accessory struct {
+	command    string
+	args       []string
+	port       string
+	healthPort string
+
+	flags func(port string) []string
+}
+
+// Run begins the accessory process. This call is not blocking.
+func (a *Accessory) Run(t *T, parentCtx context.Context) {
+	a.port, a.healthPort = getFreePort(t), getFreePort(t)
+	ctx, cancel := context.WithCancel(parentCtx)
+	t.Cleanup(func() {
+		t.Logf("cleanup: killing `%s` process", a.command)
+		cancel()
+	})
+	cmd := exec.CommandContext(ctx, a.command, append(a.args, flags(map[string]string{
+		"port":        a.port,
+		"health-port": a.healthPort,
+	})...)...)
+	t.Logf("running: %v", cmd.Args)
```

Comment:
Nit: Would make sense to also include the `command` here or not?

Reply:
It ends up as `Args[0]` actually, lol